File: 0559-maximum-depth-of-n-ary-tree/0559-maximum-depth-of-n-ary-tree.java
class Solution {

    int max = 0;

    public int maxDepth(Node root) {

        int answer = 0;
        searchDepth(root, answer);
        System.out.println("max = " + max);
        return max;
    }

    private void searchDepth(Node root, int answer) {
        if (root != null) {
            answer += 1;
            if (root.children != null) {
                int finalAnswer = answer;
                root.children.forEach(root1 -> searchDepth(root1, finalAnswer));
            }
            max = Math.max(max, answer);
        }
    }
}

File: 0559-maximum-depth-of-n-ary-tree/0559-maximum-depth-of-n-ary-tree.kt
/**
 * Definition for a Node.
 * class Node(var `val`: Int) {
 *     var children: List<Node?> = listOf()
 * }
 */

class Solution {
fun maxDepth(root: Node?): Int {
    if (root == null) return 0
    var max = 0
    for (child in root.children) {
        max = Math.max(max, maxDepth(child))
    }
    return max + 1
}
}